sentry/__init__.py

```diff
@@ -3,80 +3,99 @@
+    """ Setup an instance of :class:`sentry_sdk`. """
+    """ :param config: Sentry configuration """
 
-    :param config: Sentry configuration
-    :param client: class used to instantiate the raven client.
-    '''
     enabled = config.get('sentry_enabled', False)
-    if not (HAS_RAVEN and enabled):
+    if not (HAS_SENTRY_SDK and enabled):
         return
 
     if config.get('sentry_odoo_dir') and config.get('sentry_release'):
         _logger.debug('Both sentry_odoo_dir and sentry_release defined, choosing sentry_release')
-    options = {
-        'release': config.get('sentry_release', get_odoo_commit(config.get('sentry_odoo_dir'))),
-    }
-    for option in const.get_sentry_options():
-        value = config.get('sentry_%s' % option.key, option.default)
-        if isinstance(option.converter, collections.Callable):
-            value = option.converter(value)
-        options[option.key] = value
+
+    release = config.get('sentry_release', get_odoo_commit(config.get('sentry_odoo_dir')))
 
     level = config.get('sentry_logging_level', const.DEFAULT_LOG_LEVEL)
+
     exclude_loggers = const.split_multiple(
         config.get('sentry_exclude_loggers', const.DEFAULT_EXCLUDE_LOGGERS)
     )
+    exclude_exceptions = const.split_multiple(
+        config.get('sentry_ignore_exceptions', const.DEFAULT_IGNORED_EXCEPTIONS)
+    )
     if level not in const.LOG_LEVEL_MAP:
         level = const.DEFAULT_LOG_LEVEL
 
-    client_cls = client_cls or raven.Client
-    client = client_cls(**options)
-    handler = OdooSentryHandler(
-        config.get('sentry_include_context', True),
-        client=client,
+    sentry_logging = LoggingIntegration(
         level=const.LOG_LEVEL_MAP[level],
+        event_level=logging.WARNING
+    )
+
+    client = sentry_sdk.init(
+        dsn= config.get('sentry_dsn'),
+        environment=config.get('sentry_environment', const.DEFAULT_ENVIRONMENT),
+        release=release,
+        send_default_pii=True,
+        integrations=[sentry_logging,
+                      ThreadingIntegration(propagate_hub=True)],
+        transport=HttpTransport,
+        before_send=sanitize_odoo_processor,
     )
+
     if exclude_loggers:
-        handler.addFilter(LoggerNameFilter(
-            exclude_loggers, name='sentry.logger.filter'))
-    raven.conf.setup_logging(handler)
-    wsgi_server.application = Sentry(
-        wsgi_server.application, client=client)
+        for item in exclude_loggers:
+            ignore_logger(item)
 
-    client.captureMessage('Starting Odoo Server')
-    return client
+    if exclude_exceptions:
+        for item in exclude_exceptions:
+            ignore_logger(item)
+
+    http.root = SentryWsgiMiddleware(http.root)
 
+    with sentry_sdk.push_scope() as scope:
+        scope.set_extra('debug', True)
+        sentry_sdk.capture_message('Starting Odoo Server', 'info')
+
+    return client
 
-sentry_client = initialize_raven(odoo_config)
+initialize_sentry(odoo_config)
```

Comment:
In fact, to initialize these kind of feature we need to be sure that the module is loading in odoo

```suggestion
def post_load():
    _logger.info("Initializing sentry...")
    initialize_sentry(odoo_config)
```

And it should be defined in the hooks of the manifest:

```diff
diff --git a/sentry/__manifest__.py b/sentry/__manifest__.py
index 8ef85da1..732cd858 100644
--- a/sentry/__manifest__.py
+++ b/sentry/__manifest__.py
@@ -22,4 +22,5 @@
     'depends': [
         'base',
     ],
+    'post_load': 'post_load',
 }
```

It will fix the following error loading modules for the longpolling:

```txt
Traceback (most recent call last):                                                                                                                                                                                                                   [136/1808]
  File "/home/odoo/instance/odoo/odoo/modules/registry.py", line 60, in __new__
    return cls.registries[db_name]
  File "/home/odoo/instance/odoo/odoo/tools/func.py", line 69, in wrapper
    return func(self, *args, **kwargs)
  File "/home/odoo/instance/odoo/odoo/tools/lru.py", line 44, in __getitem__
    a = self.d[obj].me
KeyError: 'DATABASENAME'

During handling of the above exception, another exception occurred:
```

You can use another commit for that with `git commit --author="Moisés López <moylop260@vaxuoo.com>"` :)

Reply:
Done